pulp_rpm/app/models.py

```diff
@@ -501,6 +505,56 @@ def createrepo_to_dict(cls, update):
                 update, CR_UPDATE_RECORD_ATTRS.PUSHCOUNT) or ''
         }
 
+    def to_createrepo_c(self, additional_collections=[]):
+        """
+        Convert to a createrepo_c UpdateRecord object.
+
+        Args:
+            additional_collections(): additional collections to add to createrepo_c object
+
+        Returns:
+            rec(cr.UpdateRecord): createrepo_c representation of an advisory
+
+        """
+        rec = cr.UpdateRecord()
+        rec.fromstr = self.fromstr
+        rec.status = self.status
+        rec.type = self.type
+        rec.version = self.version
+        rec.id = self.id
+        rec.title = self.title
+        rec.issued_date = parse_datetime(self.issued_date)
+        rec.updated_date = parse_datetime(self.updated_date)
+        rec.rights = self.rights
+        rec.summary = self.summary
+        rec.description = self.description
+
+        for collection in self.collections.all():
+            rec.append_collection(collection.to_createrepo_c())
+
+        for collection in additional_collections:
+            rec.append_collection(collection.to_createrepo_c())
```

Comment:
I was thinking it might make more sense to have `collections` as the parameter and do:

```python
if not collections:
   collections = self.collections.all()

    for collection in collections:
        rec.append_collection(collection.to_createrepo_c())
```

Reply:
I'm not sure there is a case when I don't need collections from the advisory itself. Do you have any in mind?
For advisory merge, I'll need to combine two querysets to pass here.

Reply:
I was just thinking it was more flexible. Perhaps someone wants to to get the digest when removing collections or perhaps collections are already in memory. I would just keep it as is though and we can worry about it later.

Reply:
Ok, that's easy enough and no-risk change. I see now the reasons for the change, thanks.

Reply:
In the next push